### PR TITLE
finish routes for invitations

### PR DESCRIPTION
**Pull request by siuwafong**
I edited the Chat.js file to pull the chat and user data. 

The invitation dialog now has both a "search" tab to send invites and a "received" tab to accept or decline invites.

I am still having an issue where if I accept an invite the friend does not show up on the sidebar until I refresh the page or use the search function.

### CODE REVIEW

File: server/routes/index.js [hunk 1]
```diff
@@ -65,5 +65,76 @@ router.get("/user/:id/conversations", async (req, res, next) => {
...
+  }
+})
+
+router.post("/user/:id/sendinvite", async (req, res, next) => {
+  try {
+    const { id } = req.params
+    const invite = new Invite({
+      sender: id,
+      recipient: req.body.contactId,
+      sendDate: dayjs(),
```
Comment (sina-jamshidi) [inline server/routes/index.js:92]: We can get this information with [Javascript's built-in Date library](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#examples) without needing an external library.

File: server/routes/index.js [hunk 2]
```diff
@@ -65,5 +65,76 @@ router.get("/user/:id/conversations", async (req, res, next) => {
...
+      _id: req.body.contactId
+    }, { $addToSet: { friends: id }})
+    const newFriend = await User.find({ _id: req.body.contactId })
+    console.log(newFriend)
+    res.json(newFriend)
+  } catch (err) {
+    console.error(err.message);
+    res.status(404).send('404 Not Found');
+  }
+})
```
Comment (sina-jamshidi) [inline server/routes/index.js:137]: Just in general in the real world you would never delete this information. Data is useful to keep around for analysis or to use later somehow, like showing someone how many invites they've sent or something. So you would change the status to "accepted" and then later when you're fetching, you can ignore the accepted ones if you'd like.

File: server/routes/index.js [hunk 3]
```diff
@@ -65,5 +65,76 @@ router.get("/user/:id/conversations", async (req, res, next) => {
...
+    const { id } = req.params
+    const invite = await Invite.find({
+      sender: req.body.contactId,
+      recipient: id
+    })
+    await Invite.findOneAndDelete({ _id: invite._id})
+    await User.findByIdAndUpdate({
+      _id : id
+    }, { $addToSet: { friends: req.body.contactId }})
+    await User.findByIdAndUpdate({
```
Comment (sina-jamshidi) [inline server/routes/index.js:127]: Nice 👍

File: server/routes/index.js [hunk 4]
```diff
@@ -65,5 +65,76 @@ router.get("/user/:id/conversations", async (req, res, next) => {
...
+          { _id: { $ne: id}}
+        ]
+      })
+      res.json(matches)
+    } 
+  } catch (err) {
+    console.error(err.message);
+    res.status(404).send('404 Not Found');
+  }
+})
```
Comment (sina-jamshidi) [inline server/routes/index.js:84]: Unclear what this route is for

**Review by sina-jamshidi — COMMENTED** [review]
I realize now you hadn't shared this PR on the Slack channel so I'm not sure if it was ready to be reviewed! But I left some comments anyways. I didn't see changes to Chat.js here, but the problem you're describing sounds like the state might not be getting updated properly.

File: server/routes/index.js (hunk 4 above)
Comment (siuwafong) [reply server/routes/index.js:84]: Hi Sina. The instructions for the ticket said that I should find friends by looking up their email, so this route is to find contacts that have the email (or the substring in their email). I made a short video of the feature below.

https://www.loom.com/share/b4c56a0929ea410a940e93504e42e647

Comment (sina-jamshidi) [reply server/routes/index.js:84]: Oh okay, thanks Wilson! I misunderstood the route. It makes sense now!

**Pull request merged**